src/MarkdownTextInput.web.tsx

```diff
@@ -369,20 +371,54 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
             text = parseText(divRef.current, changedText, processedMarkdownStyle).text;
         }
 
+        const normalizedText = normalizeValue(text);
+
         if (pasteRef?.current) {
           pasteRef.current = false;
           updateSelection(e);
         }
         updateTextColor(divRef.current, text);
 
         if (onChange) {
-          const event = e as unknown as NativeSyntheticEvent<any>;
+          const event = e as unknown as NativeSyntheticEvent<{
+            count: number;
+            before: number;
+            start: number;
+          }>;
           setEventProps(event);
+
+          const newSelection = CursorUtils.getCurrentCursorPosition(divRef.current) ?? {start: 0, end: 0};
+
+          // The new text is between the prev start selection and the new end selection
+          const maybeAddedtext = normalizedText.slice(prevSelection.start, newSelection.end);
```

Comment:
How about changing this variable name to just `addedText`?

Reply:
it might be that the user has deleted text, in which case no text has been added. Hence i named it "maybe" - but we can rename it to whatever 😄